gpytorch/kernels/kernel.py

```diff
@@ -248,6 +270,8 @@ def _covar_dist(self, x1, x2, diag=False, batch_dims=None, square_dist=False, **
 
         res = None
 
+        distance_module = Distance(postprocess_func)
+
         if diag:
```

Comment:
`diag` mode should also call `postprocess`. Otherwise, the kernel has a lot of repeated code, and Kernel subclasses will have to have lots of if statements that `_covar_dist` is trying to avoid.